sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -629,11 +629,122 @@ example, if a new device is released before there is a new DPC++ release.  In
 fact, the DPC++ driver supports a command line option which allows the user
 to select an alternate configuration file.
 
-**TODO**: More information will be inserted here when we merge
-[this separate PR][7] into this design document.
+**TODO**: 
+* The names of the devices in the configuration file.
+* Location of the default device configuration file
+
+#### New features in clang compilation driver and tools
+
+NOTE: the term *device binary image* or *image* is used to refer to a device
+code form consumable by the SYCL runtime. Earlier device code forms are referred
+to as *device code module* or *device IR module*. In case of AOT, device binary
+image is a natively compiled binary, and IR module - either a SPIR-V or LLVMIR
+bitcode module.
+
+##### Overview
+After the `sycl-post-link` performs necessary aspect usage analysis and splits
+the incoming monolythic device code module into pieces - smaller device code
+modules - it outputs a file table as a result. Each row in the table corresponds
+to an individual output module, and each element of a row is a name of a file
+containing necessary information about the module, such as the code itself, its
+properties.
+
+At the action graph building stage for each requested AOT compilation target -
+SPIR-V-based (such as Gen targets) and/or non-SPIR-V-based (such as PTX) - the
+driver adds an `aspect-filter` action which filters out input file table rows
+with device code modules using features unsupported on current target. Then the
+output table goes as input into the AOT stage, and the prior filtering
+guarantees that the AOT compiler will not encounter device code it can't
+compile. In the extreme case when all device code
+modules use unsupported aspects, the input file table will be empty. The picture
+below illustrates the action graph built by the clang driver along with file
+lists and tables generated and consumed by various nodes of the graph. The
+example set of targets used for the illustration is 4 targets
+- spir64 (runtime JITted SPIR-V)
+- AOT targets
+    - non-SPIR-V based
+        - nvptx (PTX)
+    - SPIR-V based
+        - gen_11 (Intel Gen)
+        - x86_64_avx512 (AVX512)
+<br>
+<br>
+
+![Device SPIRV translation and AOT compilation](images/DeviceLinkAOTAndWrap.svg)
+
+##### Aspect filter tool
```

Comment:
I suggest renaming this tool to `sycl-feature-filter` because it will need to filter out lines in the file table based on features that a device supports, and not all of these features correspond to aspects.  For example, the `reqd_sub_group_size` feature does not have a corresponding aspect.

Reply:
The spec says that  `reqd_sub_group_size` and `reqd_work_group_size` are the only features not mapping to aspect. Also 'aspect' is more specific than 'feature', as it is defined by the SYCL spec. I would say sub/work group sizes are special cases of aspects, so aspect in the name of the tool seems more informative to me.